pkg/telemetry/statsworker.go

```diff
@@ -163,17 +236,109 @@ func coalesce(stats []*livekit.AnalyticsStat) *livekit.AnalyticsStat {
+		curStats.TotalFrames += analyticsStream.TotalFrames
+		curStats.TotalNacks += analyticsStream.TotalNacks
+		curStats.TotalPlis += analyticsStream.TotalPlis
+		curStats.TotalFirs += analyticsStream.TotalFirs
+		// add/update new video VideoLayers data to current and sum up video layer bytes/packets
+		for _, videoLayer := range analyticsStream.VideoLayers {
+			curStats.VideoLayers[videoLayer.Layer] = proto.Clone(videoLayer).(*livekit.AnalyticsVideoLayer)
+			curStats.TotalPackets += videoLayer.TotalPackets
+			curStats.TotalBytes += videoLayer.TotalBytes
+		}
+
+	}
+
+	// update currentStats
+	stats.curStats = &curStats
+
+	return
+}
+
+// find delta between  curStats and prevStats and prepare proto payload
+func (stats *Stats) computeDeltaStats() *livekit.AnalyticsStat {
 
-		streams = append(streams, stream)
+	if stats.curStats == nil {
+		return nil
+	}
+
+	// Stats in both queue/prev contain consolidated single deltaStats
+	cur := stats.curStats
+
+	var maxLayer int32
+	var maxTotalBytes uint64
+	//create a map of VideoLayers - to pick max/best layer wrt current and prev
+	curLayers := make(map[int32]*livekit.AnalyticsVideoLayer)
+	for _, layer := range cur.VideoLayers {
+		curLayers[layer.Layer] = layer
+		// identify layer which sent max data - as VideoLayers can change in current interval
+		if layer.TotalBytes > maxTotalBytes {
+			maxTotalBytes = layer.TotalBytes
+			maxLayer = layer.Layer
+		}
 	}
 
-	return &livekit.AnalyticsStat{
-		Score:   score,
-		Streams: streams,
+	// no previous stats, prepare stat
+	if stats.prevStats == nil {
+		return cur.ToAnalyticsStats(curLayers[maxLayer])
+	}
+
+	// we have prevStats, find delta between cur and prev
+	prev := stats.prevStats
+	deltaStats := Stat{}
+	deltaStats.Rtt = cur.Rtt
+	deltaStats.Jitter = cur.Jitter
+	deltaStats.TotalPlis = cur.TotalPlis - prev.TotalPlis
+	deltaStats.TotalFrames = cur.TotalFrames - prev.TotalFrames
+	deltaStats.TotalNacks = cur.TotalNacks - prev.TotalNacks
+	deltaStats.TotalFirs = cur.TotalFirs - prev.TotalFirs
+	deltaStats.TotalPacketsLost = cur.TotalPacketsLost - prev.TotalPacketsLost
+	deltaStats.TotalPrimaryPackets = cur.TotalPrimaryPackets - prev.TotalPrimaryPackets
+	deltaStats.TotalRetransmitPackets = cur.TotalRetransmitPackets - prev.TotalRetransmitPackets
+	deltaStats.TotalPaddingPackets = cur.TotalPaddingPackets - prev.TotalPaddingPackets
+	deltaStats.TotalPaddingPackets = cur.TotalPaddingPackets - prev.TotalPaddingPackets
+	deltaStats.TotalPrimaryBytes = cur.TotalPrimaryBytes - prev.TotalPrimaryBytes
+	deltaStats.TotalPaddingBytes = cur.TotalPaddingBytes - prev.TotalPaddingBytes
+	deltaStats.TotalRetransmitBytes = cur.TotalRetransmitBytes - prev.TotalRetransmitBytes
+
+	var videoLayer *livekit.AnalyticsVideoLayer
+	if len(cur.VideoLayers) > 0 {
+		videoLayer = new(livekit.AnalyticsVideoLayer)
+		// find the prev max layer to calculate frame deltas only
+		if prevMaxLayer, ok := prev.VideoLayers[maxLayer]; ok {
```

Comment:
I think we can only compare the frames when prev layer matches current layer. I think it'll have to:
1. find the current layer for the same layer id as previous, compute current round of delta with it
2. find store new max layer as prev for next round.